rollrus.go

```diff
@@ -192,50 +192,51 @@ func (r *Hook) Fire(entry *logrus.Entry) error {
 		m["time"] = entry.Time.Format(time.RFC3339)
 	}
 
+	if _, exists := m["msg"]; !exists && entry.Message != "" {
+		m["msg"] = entry.Message
+	}
+
 	if r.ignoreFunc(cause, m) {
 		return nil
 	}
 
-	return r.report(entry, cause, m, trace)
+	r.report(entry, err, m)
+
+	return nil
 }
 
-func (r *Hook) report(entry *logrus.Entry, cause error, m map[string]string, trace []uintptr) (err error) {
-	hasTrace := len(trace) > 0
+func (r *Hook) report(entry *logrus.Entry, cause error, m map[string]interface{}) {
 	level := entry.Level
 
 	r.reported = true
 
 	switch {
-	case hasTrace && level == logrus.FatalLevel:
-		_, err = r.Client.CriticalStack(cause, trace, m)
-	case hasTrace && level == logrus.PanicLevel:
-		_, err = r.Client.CriticalStack(cause, trace, m)
-	case hasTrace && level == logrus.ErrorLevel:
-		_, err = r.Client.ErrorStack(cause, trace, m)
-	case hasTrace && level == logrus.WarnLevel:
-		_, err = r.Client.WarningStack(cause, trace, m)
 	case level == logrus.FatalLevel || level == logrus.PanicLevel:
-		_, err = r.Client.Critical(cause, m)
+		skip := framesToSkip(2)
+		r.Client.ErrorWithStackSkipWithExtras(rollbar.CRIT, cause, skip, m)
```

Comment:
Same here, since this quits the program, I think there should be r.Client.Wait to make sure that the error actually reaches Rollbar.